### ISSUE TITLE
Marshal/Unmarshal behavior inconsistent with nil pointer

### ISSUE THREAD

**Issue by cArchivist** [gocarina/gocsv#175]
`Marshal*` + `Unmarshal*` should be perfect inverses of each other where possible.  In the case where a struct has a nil `*string`, the current implementation of `Marshal` writes an empty string for a nil string pointer, and `Unmarshal` exclusively outputs a pointer to an empty string.

This is a technically correct behavior, but would like an option to interpret empty values as nil if the target field is a pointer type and services the implicit inverse relationship between `Marshal` and `Unmarshal`.

**Comment by lizapolyudova** [comment]
Hey @cArchivist should I file a separate issue then for the issue described above? Or would you not consider that a bug?

**Comment by cArchivist** [comment]
When I tested the scenario from your playground link with `csv:"value,omitempty"` it worked as expected, so I closed the gap by handling variance in whitespace in #199 .  If there's an outstanding issue after #199 , please call it out, but I observed proper float pointer handling in my testing.

**Comment by lizapolyudova** [comment]
Oh my bad! Not sure how I missed the PR and the fact that it fixed the issue.
Really appreciate the turnaround! Doesn't happen every day :)
